ClinGen/clincoded
# Connects to #1286. Connects to #1308. Connects to #1321. Free text entries for experimental ontologies.

**Technical notes:**
1. For the Uberon ID format change in #1286, we also need to update any stored Uberon ID data in production database (#1337).
2. For the changes to allow both EFO and CL IDs in #1308, only the following **Experiment type** will be affected: _Functional Alteration_, _Model Systems_ and _Rescue_.
3. For the changes to allow free text option for ontologies in #1321, all **Experiment type** will be affected except _Protein Interactions_.
***
**Steps to test #1286:**
1. Go to the Experimental Data curation form page in GCI.
2. Select "Expression" as the **Experiment type**.
3. Select option **A** or **B** in the **Please select which one (A or B) you would like to curate** dropdown menu.
4. Expect to see a yellow alert box at the top of the **B. Expression** panel with the mentioned Uberon ID in the format of "UBERON:#######".
![image](https://cloud.githubusercontent.com/assets/6956310/25680980/83108602-3007-11e7-8663-171c01a4a93f.png)
5. Expect to see the placeholder text with the mentioned Uberon ID in the format of "UBERON:#######" in the **Organ of tissue relevant to disease, in which gene expression is examined in patient (Uberon ID)** input box.
![image](https://cloud.githubusercontent.com/assets/6956310/25681015/a7b7e176-3007-11e7-8db7-8615a7dc5836.png)
6. Enter "UBERON_0015228" in the **(Uberon ID)** input box and submit the form. Expect to see an error alerting the incorrect format of the ID.
![image](https://cloud.githubusercontent.com/assets/6956310/25680998/95f0e4ec-3007-11e7-92b8-a90091ca7362.png)
7. Enter "UBERON:0015228" in the **(Uberon ID)** input box and submit the form. Expect to see the form being successfully submitted.
***
**Steps to test #1308:**
1. Go to the Experimental Data curation form page in GCI.
2. Select either "Functional Alteration", "Model Systems" or "Rescue" as the **Experiment type**.
3. Select "Engineered equivalent" as in the **Patient cells with candidate mutation or engineered equivalent?** dropdown menu in "Functional Alteration". Similarly, select "Engineered equivalent" as in the **Non-human animal or cell-culture model?** dropdown menu in "Model Systems", or the **Patient cells with or engineered equivalent?** dropdown in "Rescue".
4. Expect to see a yellow alert box above the **(EFO or CL ID)** input box, with the mentioning of both EFO and Cell Ontology (CL) terms.
![image](https://cloud.githubusercontent.com/assets/6956310/25681893/dbe1ca22-300a-11e7-8759-770955a4db70.png)
5. Expect to see the "Search the EFO or Cell Ontology (CL) using the OLS (Ontology Lookup Service)." text right above the **(EFO or CL ID)** input box. Click on the individual links among this line of text and expect to be taken to the intended destination pages.
![image](https://cloud.githubusercontent.com/assets/6956310/25681902/e76f6e80-300a-11e7-8cbc-2a1878fe9e57.png)
6. Expect to see the placeholder text describing both EFO and CL IDs to be allowed in the **Engineered equivalent cell type/line (EFO or CL ID)** input box.
7. Enter "HP:0010704" in the **(EFO or CL ID)** input box and submit the form. Expect to see an error alerting the incorrect entry of the ID.
![image](https://cloud.githubusercontent.com/assets/6956310/25681981/3e5aa4f8-300b-11e7-8469-9a28c51edb22.png)
8. Enter "EFO_0001187" or "CL_0000057" in the **(EFO or CL ID)** input box and submit the form. Expect to see the form being successfully submitted.
***
**Steps to test #1321:**
1. Go to the Experimental Data curation form page in GCI.
2. Select "Biochemical Function" as the **Experiment type**.
3. Select option **A** or **B** in the **Please select which one (A or B) you would like to curate** dropdown menu.
4. Expect to see the _Required_ **Identified function of gene in this record (free text)** input box right below the _Required_ **Identified function of gene in this record (GO ID)** input box.
![image](https://cloud.githubusercontent.com/assets/6956310/25682554/790a0dbc-300d-11e7-86e9-9554525ef092.png)
5. Submit the form without entering any value in both of the described input boxes. Expect to see an error alerting the incorrect entries in these boxes.
![image](https://cloud.githubusercontent.com/assets/6956310/25682647/ca366fdc-300d-11e7-8bfc-65d81d07e746.png)
6. Enter some free text in the **Identified function of gene in this record (free text)** input box. Expect to see the **Identified function of gene in this record (GO ID)** input box being no longer _Required_. And now expect to be able submit the form successfully.
7. The steps described above can be used to test the other ontologies:
a. Uberon ID in **Expression**.
b. GO, CL and EFO (or CL) IDs in **Functional Alteration**.
c. EFO (or CL) ID in **Model Systems**.
c. CL and EFO (or CL) ID in **Rescue**.
